packages/relay/src/lib/clients/cache/impl/customLRUCache.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ *
+ * Hedera JSON RPC Relay
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import LRUCache from 'lru-cache';
+import findConfig from 'find-config';
+import fs from 'fs';
+import { Logger } from 'pino';
+import { SpendingPlanConfig } from '../../../types/spendingPlanConfig';
+import { EthAddressHbarSpendingPlanRepository } from '../../../db/repositories/hbarLimiter/ethAddressHbarSpendingPlanRepository';
+import { IPAddressHbarSpendingPlanRepository } from '../../../db/repositories/hbarLimiter/ipAddressHbarSpendingPlanRepository';
+import { HbarSpendingPlanRepository } from '../../../db/repositories/hbarLimiter/hbarSpendingPlanRepository';
+
+export class CustomLRUCache<K, V> extends LRUCache<K, V> {
```

Comment:
I think it would make sense to restrict the generic type K to a string.  Later in the code there is a test to check if key is of type string.
```
export class CustomLRUCache<K extends string, V> extends LRUCache<K, V> {
```